lib/ecto/uuid.ex

```diff
@@ -100,9 +100,7 @@ defmodule Ecto.UUID do
 
   # Callback invoked by autogenerate fields.
   @doc false
-  def autogenerate do
-    %Ecto.Query.Tagged{type: :uuid, value: bingenerate()}
-  end
+  def autogenerate, do: generate()
```

Comment:
Unfortunately this is expensive because we generate the binary to become a UUID and then to become a binary again.